packages/server/src/profile/profile.controller.ts

```diff
@@ -218,27 +222,25 @@ export class ProfileController {
           ERROR_MESSAGES.profileController.noDiskSpace,
         );
       }
-      const fileName =
-        user.id +
-        '-' +
-        Math.random().toString(36).substring(2, 15) +
-        Math.random().toString(36).substring(2, 15);
+      const fileName = user.id + '-' + Date.now().toString() + '.webp';
+
+      // Create the upload location if it doesn't exist
       if (!fs.existsSync(process.env.UPLOAD_LOCATION)) {
         fs.mkdirSync(process.env.UPLOAD_LOCATION, { recursive: true });
       }
 
       const targetPath = path.join(process.env.UPLOAD_LOCATION, fileName);
 
       try {
-        const image = await Jimp.read(file.buffer);
-        image.resize(256, Jimp.AUTO);
-        await image.writeAsync(targetPath);
+        await sharp(file.buffer).resize(256).webp().toFile(targetPath);
```

Comment:
Maybe let's try resizing it to 128. This may make it a little blurry on the profile page on mobile but otherwise I think should be fine

Reply:
![Screenshot 2024-10-02 111727](https://github.com/user-attachments/assets/39b6ca72-d74b-4bad-aef4-630d6c00517b)
![Screenshot 2024-10-02 111756](https://github.com/user-attachments/assets/34789a63-bb2e-4e3d-90d7-d8e2e62f5bd2)

I think that it might be an unnecessary optimization that would hinder the visual experience of the application. I think resizing the organization logo makes more sense as there are no instances where it is displayed as any more detailed than 48x48, but that's not the case with the profile photos.

Reply:
The experience of changing profile photos is especially affected because the user goes from a high-res version of their image in the cropper to an instantly low-res version on display within the profile page.